core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/printing/Backend.java

```diff
@@ -98,9 +68,69 @@ public boolean print(byte[] content, JsonNode configuration) {
       log.error(buffer.toString());
       return true;
     }
+  },
+  RAW_TCP("host", "port") {
+
+    @Override
+    public boolean print(byte[] content, JsonNode configuration) {
+      String host = configuration.get("host").asText();
+      int port = configuration.get("port").asInt();
+      try (Socket socket = new Socket(host, port)) {
+        socket.getOutputStream().write(content);
+      } catch (IOException e) {
+        log.error("Failed to print to socket", e);
+        return false;
+      }
+      return true;
+    }
+
+  },
+  ZEBRA_FTP("host", "password") {
+    @Override
+    public boolean print(byte[] content, JsonNode configuration) {
+      return sendFtpFile(configuration.get("host").asText(), "admin", configuration.get("password").asText(),
+          null, Arrays.hashCode(content) + ".ZPL", content);
+    }
+
   };
   private static final Logger log = LoggerFactory.getLogger(Backend.class);
 
+  protected static boolean sendFtpFile(String host, String username, String password, String directory, String filename, byte[] content) {
+
+    try {
+      FTPClient ftp = TransmissionUtils.ftpConnect(host, username, password);
+      if (ftp == null || !ftp.isConnected()) {
+        log.error("FTP client isn't connected. Please supply a client that has connected to the host.");
+        return false;
+      }
+
+      if (directory == null) {// Do Nothing
+
+      } else if (!ftp.changeWorkingDirectory(directory)) {
+        log.error("Desired path does not exist on the server");
+        ftp.logout();
+        ftp.disconnect();
```

Comment:
Could put logout+disconnect in a finally block connected to the try block that wraps most of this method